### PR TITLE
More flexible mediasite regex to download public videos

### PR DESCRIPTION
**Pull request by Exteris**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Make the Mediasite download regex slightly more flexible.

### CODE REVIEW

File: youtube_dl/extractor/mediasite.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 
 class MediasiteIE(InfoExtractor):
-    _VALID_URL = r'(?xi)https?://[^/]+/Mediasite/Play/(?P<id>[0-9a-f]{32,34})(?P<query>\?[^#]+|)'
+    _VALID_URL = r'(?xi)https?://[^/]+/Mediasite/(Play|Showcase/livebroadcast/Presentation)/(?P<id>[0-9a-f]{32,34})(?P<query>\?[^#]+|)'
```
Comment (dstftw) [inline youtube_dl/extractor/mediasite.py]: Don't capture groups you don't use. Add only matching test.

Comment (Exteris) [reply youtube_dl/extractor/mediasite.py]: I'm having some issues due to an unrelated bug/change in mediasite?

```ERROR: [Mediasite]  is not a supported codec; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/media/data/Projects/youtube-dl/youtube_dl/YoutubeDL.py", line 793, in extract_info
    ie_result = ie.extract(url)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 508, in extract
    ie_result = self._real_extract(url)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/mediasite.py", line 183, in _real_extract
    fatal=False))
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 2279, in _extract_ism_formats
    return self._parse_ism_formats(ism_doc, urlh.geturl(), ism_id)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 2306, in _parse_ism_formats
    self.report_warning('%s is not a supported codec' % fourcc)
  File "/media/data/Projects/youtube-dl/youtube_dl/extractor/common.py", line 888, in report_warning
    '[%s] %s%s' % (self.IE_NAME, idstr, msg))
  File "/media/data/Projects/youtube-dl/test/helper.py", line 245, in _report_warning
    real_warning(w)
  File "test/test_download.py", line 52, in report_warning
    raise ExtractorError(message)
youtube_dl.utils.ExtractorError: [Mediasite]  is not a supported codec; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

This occurs for all other tests too.

Comment (dstftw) [reply youtube_dl/extractor/mediasite.py]: Add `only_matching` test.

Comment (Exteris) [reply youtube_dl/extractor/mediasite.py]: > Add `only_matching` test.

I'm sorry I don't really understand what you mean...

**Pull request merged**